splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -238,6 +238,10 @@ private SdkTracerProvider buildTracerProvider(
                 .setResource(resource);
         initializationEvents.add(new RumInitializer.InitializationEvent("tracerProviderBuilderInitialized", timingClock.now()));
 
+        if (config.isSessionBasedSamplerEnabled()) {
+            tracerProviderBuilder.setSampler(Sampler.parentBased(new SessionIdRatioBasedSampler(config.getSessionBasedSamplerRatio(), sessionId)));
```

Comment:
Is there any reason to use `parentBased` here? I think the only input is the session id, and that should be available for all spans, yes?

Reply:
The reason is that if there is a parent span created by the user manually, which is not sampled (sampling is only applied to the tracer used internally, not any tracer created by the user), that would cause the situation where a single trace contains some spans which are sampled and some which are not, which is generally undesirable. The expected behavior is that a trace is either wholly sampled or not. Parent-based makes sure to copy the sampled bit from the parent if a parent exists.

Reply:
Removed `parentBased` around it.